### PR TITLE
Adds support for as()

### PR DESCRIPTION
**Pull request by ike709**
Doing 
```
for(var/item/thing/T as() in some_list)
``` 
is the same as 
```
for(var/item/thing/T as anything in some_list)
```

Both define a loop without internal typechecking.

### CODE REVIEW

**Review by ZeWaka — APPROVED** [review]

File: OpenDreamShared/Compiler/DM/DMParser.cs [hunk 1]
```diff
@@ -1633,8 +1633,13 @@ class DMParser : Parser<Token> {
...
 
             if (Check(TokenType.DM_As)) {
+                if (Check(TokenType.DM_LeftParenthesis) && Check(TokenType.DM_RightParenthesis))
+                {
+                    return DMValueType.Anything;
+                }
+                
                 Whitespace();
-
+
```
Comment (ike709) [inline OpenDreamShared/Compiler/DM/DMParser.cs:1642]:
```suggestion
```

File: OpenDreamShared/Compiler/DM/DMParser.cs [hunk 2]
```diff
@@ -1633,8 +1633,12 @@ class DMParser : Parser<Token> {
             DMValueType type = DMValueType.Anything;
 
             if (Check(TokenType.DM_As)) {
+                if (Check(TokenType.DM_LeftParenthesis) && Check(TokenType.DM_RightParenthesis))
```
Comment (wixoaGit) [inline OpenDreamShared/Compiler/DM/DMParser.cs:1636]: Whitespace needs to be checked for before the left parenthesis, and `Consume` should be used for the right parenthesis

**Review by wixoaGit — CHANGES_REQUESTED** [review]
This works for `as()` but doesn't work if you give it any types, like `as(num|text)`.

**Comment by ike709** [comment]
Didn't know you could insert types like that, should be good now.

**Comment by ike709** [comment]
This works against beestation. `as()` and `as(text|num)` are valid but `as(` errors as expected. `as text|num` seems unaffected

**Review by wixoaGit — APPROVED** [review]

**Pull request merged**